feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go

```diff
@@ -0,0 +1,617 @@
+package p4rt_election_test
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cisco-open/go-p4/p4rt_client"
+	"github.com/cisco-open/go-p4/utils"
+	"github.com/openconfig/featureprofiles/feature/experimental/p4rt/internal/p4rtutils"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygot/ygot"
+	p4_v1 "github.com/p4lang/p4runtime/go/p4/v1"
+)
+
+// Variable definitions
+var (
+	streamName   = "p4rt"
+	p4InfoFile   = flag.String("p4info_file_location", "../../wbb.p4info.pb.txt", "Path to the p4info file.")
+	p4rtNodeName = flag.String("p4rt_node_name", "SwitchChip1/0", "component name for P4RT Node")
+	gdpEtherType = *ygot.Uint32(0x6007)
+	portID       = *ygot.Uint32(10)
+	deviceID     = *ygot.Uint64(1)
+	highID       = *ygot.Uint64(0)
+	pDesc        = "Primary Connection"
+	sDesc        = "Secondary Connection"
+	npDesc       = "New Primary"
+)
+
+// Define PacketIO Interface
+type PacketIO interface {
+	GetTableEntry(delete bool) []*p4rtutils.ACLWbbIngressTableEntryInfo
+}
+
+// Define GDP Packet io structure
+type GDPPacketIO struct {
+	PacketIOPacket
+	IngressPort string
+}
+
+// Define PacketIO Structure
+type PacketIOPacket struct {
+	EthernetType *uint32
+}
+
+// Define Multiple Client Args Structure
+type mcTestArgs struct {
+	name       string
+	Items      [2]*testArgs
+	expectPass bool
+}
+
+// Define Test Args Structure
+type testArgs struct {
+	desc              string
+	lowID             uint64
+	highID            uint64
+	deviceID          uint64
+	handle            *p4rt_client.P4RTClient
+	expectStatus      int32
+	expectFinalStatus int32
+	expectPass        bool
+	expectWrite       bool
+	expectRead        bool
+}
+
+// configureDeviceId configures p4rt device-id on the DUT.
+func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
+	component := oc.Component{}
+	component.IntegratedCircuit = &oc.Component_IntegratedCircuit{}
+	component.Name = ygot.String(*p4rtNodeName)
+	component.IntegratedCircuit.NodeId = ygot.Uint64(deviceID)
+	gnmi.Replace(t, dut, gnmi.OC().Component(*p4rtNodeName).Config(), &component)
+}
+
+// configurePortId configures p4rt port-id on the DUT.
+func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	portName := dut.Port(t, "port1").Name()
+	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portID)
+}
+
+// Create client connection
+func clientConnection(t *testing.T, dut *ondatra.DUTDevice) *p4rt_client.P4RTClient {
+	clientHandle := p4rt_client.P4RTClient{}
+	if err := clientHandle.P4rtClientSet(dut.RawAPIs().P4RT().Default(t)); err != nil {
+		t.Fatalf("Could not initialize p4rt client: %v", err)
+	}
+	return &clientHandle
+}
+
+// Setup P4RT Arbitration Stream
```

Comment:
Can we define here what the return argument is? i.e., the arbitration status code.

Reply:
@robshakir , did you mean the return on line 93 or the comment from the function starting on line 96?